lib/player.js

```diff
@@ -2837,6 +2837,32 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     return this.config_;
   }
 
+  /**
+   * Returns the ratio of video length buffered compared to buffering Goal
+   * @return {number}
+   * @export
+   */
+  getBufferFullness() {
+    const bufferedLength = this.video_ ?
+        this.video_.buffered.length : 0;
+    const bufferedEnd =
+        bufferedLength ? this.video_.buffered.end(bufferedLength - 1) : 0;
+    const bufferingGoal = this.getConfiguration().streaming.bufferingGoal;
+    const lengthToBeBuffered = this.video_ ? Math.min(this.video_.currentTime +
+        bufferingGoal, this.seekRange().end) : 0;
+
+    if (this.video_) {
```

Comment:
Since this.video_ is needed here, maybe you should check for it at the top and just return 0 early if this.video_ is null.  Then, you can clean up all the early calculations to remove the repeated ternary checks.